src/dev/run_check_core_api_changes.ts

```diff
@@ -0,0 +1,245 @@
+    },
+    unknown(name) {
+      extraFlags.push(name);
+      return false;
+    },
+  });
+
+  if (extraFlags.length > 0) {
+    for (const flag of extraFlags) {
+      log.error(`Unknown flag: ${flag}`);
+    }
+
+    opts.help = true;
+  }
+
+  if (opts.help) {
+    process.stdout.write(
+      dedent(chalk`
+        {dim usage:} node scripts/check_core_api_changes [...options]
+
+        Checks for any changes to the Kibana Core API
+
+        Examples:
+
+          {dim # Checks for any changes to the Kibana Core API}
+          {dim $} node scripts/check_core_api_changes
+
+          {dim # Checks for any changes to the Kibana Core API and updates the documentation}
+          {dim $} node scripts/check_core_api_changes --docs
+
+          {dim # Checks for and automatically accepts and updates documentation for any changes to the Kibana Core API}
+          {dim $} node scripts/check_core_api_changes --accept
+
+        Options:
+          --accept    {dim Accepts all changes by updating the API Review files and documentation}
+          --docs      {dim Updates the Core API documentation}
+          --help      {dim Show this message}
+      `)
+    );
+    process.stdout.write('\n');
+    return !(extraFlags.length > 0);
+  }
+
+  log.info(`Core ${folder} API: checking for changes in API signature...`);
+
+  await runBuildTypes().catch(e => {
+    if (e) {
+      log.error(e);
+      return false;
+    }
+  });
+
+  const { apiChanged, warnings, errors } = runApiExtractor(folder, opts.accept);
+  await renameExtractedApiPackageName(folder);
+
+  const apiReviewFilePath =
+    apiExtractorConfig(folder)!.apiReviewFile!.apiReviewFolder + 'kibana.api.md';
+
+  if (apiChanged && opts.accept) {
+    log.warning(`You have changed the public signature of the ${folder} Core API`);
+    log.warning(
+      `Please commit the updated API documentation and the review file in '${apiReviewFilePath}' \n`
+    );
+  }
+
+  if (apiChanged && !opts.accept) {
+    log.warning('You have changed the public signature of the Kibana Core API');
+    log.warning(
+      'To accept these changes run `node scripts/check_core_api_changes.js --accept` and then:\n' +
+        `\t 1. Commit the updated documentation and API review file ${apiReviewFilePath}' \n` +
+        "\t 2. Describe the change in your PR including whether it's a major, minor or patch"
+    );
+  }
+
+  if (!apiChanged) {
+    log.info(`Core ${folder} API: no changes detected ✔`);
+  }
+
+  if (opts.accept || opts.docs) {
+    await runApiDocumenter(folder)
+      .then(() => {
+        log.info(`Core ${folder} API: updated documentation ✔`);
+      })
+      .catch(e => {
+        log.error(e);
+        return false;
+      });
+  }
+
+  // If the API changed and we're not accepting the changes, exit process with error
+  if (apiChanged && !opts.accept) {
+    return false;
+  }
+
+  // If any errors or warnings occured, exit with an error
+  if (errors.length > 0 || warnings.length > 0) {
```

Comment:
api-extractor gives warnings for problems like "Warning: (ae-internal-missing-underscore) The name bootstrap should be prefixed with an underscore because the declaration is marked as "@internal"" It's possible to silence some of the rules in the latest beta releases, but I found it very helpful to be able to inspect the generated documentation, even if there were warnings present that would break the build.

Once api-extractor v7 is released we can clean this up a lot.